client/e2e/integration/user-list.spec.ts

```diff
@@ -0,0 +1,135 @@
+import { UserListPage } from '../support/user-list.po';
+
+const page = new UserListPage();
+
+describe('User list', () => {
+
+  beforeEach(() => {
+    page.navigateTo();
+  });
+
+  it('Should have the correct title', () => {
+    page.getUserTitle().should('have.text', 'Users');
+  });
+
+  it('Should type something in the name filter and check that it returned correct elements', () => {
+    // Filter for user 'Lynn Ferguson'
+    cy.get('#user-name-input').type('Lynn Ferguson');
+
+    // All of the user cards should have the name we are filtering by
+    page.getUserCards().each(e => {
+      cy.wrap(e).find('.user-card-name').should('have.text', 'Lynn Ferguson');
+    });
+
+    // (We check this two ways to show multiple ways to check this)
+    page.getUserCards().find('.user-card-name').each($el =>
+      expect($el.text()).to.equal('Lynn Ferguson')
+    );
+  });
+
+  it('Should type something in the company filter and check that it returned correct elements', () => {
+    // Filter for company 'OHMNET'
+    cy.get('#user-company-input').type('OHMNET');
+
+    // All of the user cards should have the company we are filtering by
+    page.getUserCards().find('.user-card-company').each($card => {
+      cy.wrap($card).should('have.text', 'OHMNET');
+    });
+  });
+
+  it('Should type something partial in the company filter and check that it returned correct elements', () => {
+    // Filter for companies that contain 'ti'
+    cy.get('#user-company-input').type('ti');
+
+    // Go through each of the cards that are being shown and get the companies
+    page.getUserCards().find('.user-card-company')
+      // We should see these companies
+      .should('contain.text', 'MOMENTIA')
+      .should('contain.text', 'KINETICUT')
+      // We shouldn't see these companies
+      .should('not.contain.text', 'DATAGENE')
+      .should('not.contain.text', 'OHMNET');
+  });
+
+  it('Should type something in the age filter and check that it returned correct elements', () => {
+    // Filter for users of age '27'
+    cy.get('#user-age-input').type('27');
+
+    // Go through each of the cards that are being shown and get the names
+    page.getUserCards().find('.user-card-name')
```

Comment:
I think the issue with that was that the user list doesn't ever display ages.

Reply:
Ah, and since this is all E2E, if it's not visible in the UI we can't talk about it. So checking names may be the best thing we can do without changing the UI. It seems like we'd want a comment to that effect?